src/util/templates/StaticTemplateEngine.ts

```diff
@@ -0,0 +1,23 @@
+import type { AsyncHandler } from '../handlers/AsyncHandler';
+import type { TemplateEngineInput, Template } from './TemplateEngine';
+import { TemplateEngine } from './TemplateEngine';
+import Dict = NodeJS.Dict;
+
+export class StaticTemplateEngine<T extends Dict<any> = Dict<any>> extends TemplateEngine<T> {
+  private readonly template: Template;
+  private readonly templateEngine: AsyncHandler<TemplateEngineInput<T>, string>;
+
+  public constructor(templateEngine: AsyncHandler<TemplateEngineInput<T>, string>, template: Template) {
+    super();
+    this.template = template;
+    this.templateEngine = templateEngine;
+  }
+
+  public async handle({ contents, template }: TemplateEngineInput<T>): Promise<string> {
+    if (typeof template !== 'undefined') {
+      throw new Error('StaticTemplateEngine does not support template as handle input, ' +
+          'provide a template via the constructor instead!');
+    }
```

Comment:
This should be in the `canHandle` call. It should also check `this.templateEngine.canHandle` there, and then only call `this.templateEngine.handle` here.